Ginger/MSAccessDB/MSAccessDBCon.cs

```diff
@@ -0,0 +1,352 @@
+                    ReturnList.Add(Records);
+                }
+            }
+            catch (Exception e)
+            {
+                Reporter.ToLog(eLogLevel.ERROR, "Failed to execute query:" + Query, e);
+                throw e;
+            }
+            finally
+            {
+                if (reader != null)
+                    reader.Close();
+            }
+            return ReturnList;
+        }
+       
+        public string GetConnectionString(Dictionary<string,string> parameters)
+        {
+            string connStr = null;
+            bool res;
+            res = false;
+
+            string ConnectionString= parameters.FirstOrDefault(pair => pair.Key == "ConnectionString").Value;
+            string User= parameters.FirstOrDefault(pair => pair.Key == "UserName").Value;
+            string Password = parameters.FirstOrDefault(pair => pair.Key == "Password").Value;
+            string TNS = parameters.FirstOrDefault(pair => pair.Key == "TNS").Value;
+
+            if (String.IsNullOrEmpty(ConnectionString) == false)
+            {
+                connStr = ConnectionString.Replace("{USER}", User);
+
+                String deCryptValue = EncryptionHandler.DecryptString(Password, ref res, false);
+                if (res == true)
+                { connStr = connStr.Replace("{PASS}", deCryptValue); }
+                else
+                { connStr = connStr.Replace("{PASS}", Password); }
+            }
+            else
+            {
+                String strConnString = TNS;
+                String strProvider;
+                connStr = "Data Source=" + TNS + ";User Id=" + User + ";";
+
+                String deCryptValue = EncryptionHandler.DecryptString(Password, ref res, false);
+
+                if (res == true) { connStr = connStr + "Password=" + deCryptValue + ";"; }
+                else { connStr = connStr + "Password=" + Password + ";"; }
+
+
+                if (strConnString.Contains(".accdb"))
+                {
+                    strProvider = "Provider=Microsoft.ACE.OLEDB.12.0;";
+                }
+                else { strProvider = "Provider=Microsoft.ACE.OLEDB.12.0;"; }
+
+                connStr = strProvider + connStr;
+            }
+            return connStr;
+        }
+
+        public string GetSingleValue(string Table, string Column, string Where)
+        {
+            string sql = "SELECT {0} FROM {1} WHERE {2}";
+            sql = String.Format(sql, Column, Table, Where);
+            String rc = null;
+            DbDataReader reader = null;
+            if (MakeSureConnectionIsOpen())
+            {
+                try
+                {
+                    DbCommand command = conn.CreateCommand();
+                    command.CommandText = sql;
+                    command.CommandType = CommandType.Text;
+
+                    // Retrieve the data.
+                    reader = command.ExecuteReader();
+                    while (reader.Read())
```

Comment:
if we know we will read only one then remove the while

Reply:
Done